static/js/state.ts

```diff
@@ -0,0 +1,143 @@
+import { ClientMessages } from './client-messages';
+
+/**
+ * The global state we keep
+ *
+ * This is in desperate need of cleaning up!!
+ *
+ * This shouldn't contain as much as it does.
+ */
+export interface State {
+  /**
+   * Current language
+   *
+   * Written: by every page, on page load.
+   * Used: on the code page, to do speech synthesis and to send to the server.
+   */
+  lang: string;
+
+  /**
+   * Current level
+   *
+   * Written: by every page, on page load.
+   *
+   * Used: on the code page, to initialize the highlighter, to translate the program,
+   * to determine timeouts, to load the quiz iframe, to show the variable inspector,
+   * to show a debugger,  to load parsons exercises, to initialize a default save name.
+   */
+  level: number;
+
+  /**
+   * Current keyword language
+   *
+   * Written: by every page, on page load.
+   *
+   * Used: set on the Ace editor, and then is used to do some magic that I don't
```

Comment:
Maybe @TiBiBa can help us here?

Reply:
If I recall correctly we did use to set a keyword language on each ace editor. This is from when we supported the dynamic keyword switcher within the editor. Enabling us to support a separate keyword language on the main editor and the adventure editors at the same time. 

The last time I worked on this code we decided to keep the switcher available to the admin. So if that is still the case we should keep it, it might still be used to determine the highlighter language?

Reply:
Yes, we support the feature to translate programs from the front-end, but it is only available to me, to show in demos, kids don't really need it.